Framework/PythonInterface/plugins/algorithms/MaskAngle.py

```diff
@@ -42,61 +44,56 @@ def PyInit(self):
                              mantid.kernel.StringListValidator(['TwoTheta', 'Phi']),
                              'Which angle to use')
         self.declareProperty(mantid.kernel.IntArrayProperty(name="MaskedDetectors", direction=mantid.kernel.Direction.Output),
-                             doc="List of detector masked, with scatterin angles between MinAngle and MaxAngle")
+                             doc="List of detector masked, with scattering angles between MinAngle and MaxAngle")
 
     def validateInputs(self):
         issues = dict()
-
         ws = self.getProperty("Workspace").value
-
-        try:
-            if type(ws).__name__ == "WorkspaceGroup":
-                for w in ws:
-                    w.getInstrument().getSource().getPos()
-            else:
-                ws.getInstrument().getSource().getPos()
-        except (RuntimeError, ValueError, AttributeError, TypeError):
+        hasInstrument = True
+        if type(ws).__name__ == "WorkspaceGroup" and len(ws) > 0:
+            for item in ws:
+                hasInstrument = hasInstrument and len(item.componentInfo()) > 0
+        else:
+            hasInstrument = len(ws.componentInfo()) > 0
+        if not hasInstrument:
             issues["Workspace"] = "Workspace must have an associated instrument."
-
         return issues
 
+    def _get_phi(self, det_index, detector_info):
+        pos = detector_info.position(det_index)
+        # see Detector.cpp for phi definition
```

Comment:
Note that that phi definition is wrong in general, which is the reason we do not have it in `DetectorInfo`: It assumes that the beam is aligned with Z (which is true for most instruments currently, as far as I know, but not for those used in unit tests, I think). Basically this "phi" is the phi in a spherical coordinate system, but in neutron scattering I understand it is usually understood as the azimuth of the scattering process.

Reply:
Maybe add a comment here that this is strictly speaking a bug?